### PR TITLE
Add A lot of event

### PR DESCRIPTION
**Pull request by louis1706**

### CODE REVIEW

**Comment by louis1706** [comment]
I have not finish

File: Exiled.Events/Handlers/Player.cs
Comment (louis1706) [inline Exiled.Events/Handlers/Player.cs:413]: It's an breaking change

File: Exiled.Events/Handlers/Player.cs
Comment (louis1706) [inline Exiled.Events/Handlers/Player.cs:418]: It's an breaking change

**Comment by joker-119** [comment]
Breaking changes need to be marked with the [Obsolete(string)] attribute, not just removed.

**Comment by louis1706** [comment]
- I would ask for Obsolete  if i have done it correctly (and if not how i should done it)
- and if all the naming it's good
- and comment certainly really cringe

File: Exiled.Events/Handlers/Player.cs
Comment (joker-119) [reply Exiled.Events/Handlers/Player.cs:418]: "Use Handlers.Scp330.InteractingScp330"

**Review by joker-119 — CHANGES_REQUESTED** [review]
I'll convert all the prefixes to transpilers later before merging this.

**Comment by louis1706** [comment]
> I'll convert all the prefixes to transpilers later before merging this.

Nao going to make an API for Scp244 or/and Scp330 i thinks it's preferable for wait

**Comment by NaoUnderscore** [comment]
> > I'll convert all the prefixes to transpilers later before merging this.
> 
> Nao going to make an API for Scp244 or/and Scp330 he say it's preferable to wait

Go ahead, Scp244 doesn't need an API, we already have a Scp244 class, I thought we didn't have it 🤷🏼‍♂️

**Comment by Undid-Iridium** [comment]
@joker-119 @NaoUnderscore FYI, I am attempting to translate the prefix's into transpilers. https://github.com/louis1706/EXILED/pull/5. If you have already done it let me know but I will be doing that between today and tomorrow. Thanks

**Comment by louis1706** [comment]
With the prefix work fine The Transpiler are done by @Undid-Iridium

**Comment by Undid-Iridium** [comment]
Update: I am still working on it, I made the assumption the prefixes were correct which is not the case. I am done testing 4/6 (One of which I cannot fix (I wrote in #dev-talk but Scp244Complete gets terminated somehow (Gameobject deletion? Session Pipe terminated?))). I will test the other two transpilers tomorrow after reviewing the original prefixes and their intended functionality. https://github.com/louis1706/EXILED/pull/6 will continue to have more updates regarding the changes being done. If I get anything wrong, let me know. Thanks

File: Exiled.Events/EventArgs/PickingUpScp244EventArgs.cs
Comment (louis1706) [inline Exiled.Events/EventArgs/PickingUpScp244EventArgs.cs:19]: Miss pickup

File: Exiled.Events/EventArgs/DroppingUpScp330EventArgs.cs
Comment (louis1706) [inline Exiled.Events/EventArgs/DroppingUpScp330EventArgs.cs:20]: DroppingScp330 instead of DroppingUpScp330

File: Exiled.Events/EventArgs/DroppingUpScp330EventArgs.cs
Comment (louis1706) [inline Exiled.Events/EventArgs/DroppingUpScp330EventArgs.cs:31]: Scp330 instead of giving Item

**Review by louis1706 — COMMENTED** [review]
Reminder (i will change this tomorrow)

**Comment by Undid-Iridium** [comment]
I will say that the transpilers are not cleanup yet, especially the nops. However Yamato and I have tested 5/6 of the transpiler changes to work as intended, as far as we can tell. Cleanup can be done from this point forward. @NaoUnderscore is going to work on the last one, for Scp330SeverInteract, the transpiler did work for everything except us trying to overwrite the ServerPickupProcess call. There are several solution around this but I wanted to work with references, and it did not work. Hopefully Nao has better luck. Thanks again to @BuildBoy12 @NaoUnderscore for trying to help, and @louis1706 for proof reading/updating my incorrect variable/class names. Thanks

**Comment by NaoUnderscore** [comment]
Go ahead with Pre/Postfixes, I'll update them to IL once a stable and fixed Exiled version is released.

**Comment by louis1706** [comment]
> Go ahead with Pre/Postfixes, I'll update them to IL once a stable and fixed Exiled version is released.

Okay

**Comment by Undid-Iridium** [comment]
I was able to figure out why the transpiler call to static function with reference was not working. I am going to test it, and then make a pr again on Yamato's branch. Thereafter I will do cleanup. If Nao wants to rewrite/go through and fix/clean it up that's fine. Up to you all

**Comment by Undid-Iridium** [comment]
@NaoUnderscore Louis and I did testing today and are 90%+ confident of the changes, but I have a pr for https://github.com/louis1706/EXILED/pull/9 for some extra cleanup. In addition, I want to try to resolve a generic function call I worked around with methodbuilder. You can take a look if you like, thanks

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]
Fix grammar, there're some typos somewhere.

File: Exiled.Events/Patches/Events/Scp244/PickingUpScp244.cs
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp244/PickingUpScp244.cs:74]: Do not use Ret + Nop.

File: Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs:47]: Don't really care, yknow, this is not Exiled Playground.

File: Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs:86]: Use short form when and where it's possible.

File: Exiled.Events/Patches/Events/Scp244/UsingScp244.cs
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp244/UsingScp244.cs:42]: Same as above.

File: Exiled.Events/Patches/Events/Scp244/UsingScp244.cs
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp244/UsingScp244.cs:77]: Short form required.

File: Exiled.Events/Patches/Events/Scp244/UsingScp244.cs
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp244/UsingScp244.cs:80]: Don't use Ret + Nop.

File: Exiled.Events/Patches/Events/Scp330/DroppingCandy.cs
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp330/DroppingCandy.cs:53]: Same as above.

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]
I don't feel like reviewing all the patches, because there're the same issues as the reviewed ones. Also, I don't really find necessary describing and/or commenting every IL step as it's probably made from and for who's writing them. Counting the stack is useless, because you'd have to maintain all the comments, this means that you have to update transpilers when needed, since SL-12.0 is not so far away, and you also have to update comments in order to make them coherent with the written code.
Since I'll be the one who's gonna update them, I don't want to lose my time at updating comments made by someone which is clearly over-commenting everything.
That said, it's not useless commenting code, but that's an over-commenting case, which is bad.
You're also breaking style patterns by adding curly brackets in the `for` loops where you're `yield return`ing the new instructions. Let's just keep everything as more uniform as possible, right?
Commenting IL code is a double-edged sword, because you should only comment logical steps rather than talking about "original {INSERT_SOMETHING_HERE} logic" or something like that, because as I said before, we're not here to maintain a book written by Shakespeare; maintaining the code is already enough in my opinion.
I appreciate the fact that you're documenting IL, which is needed sometimes, but it's not the right way to document IL; you're not writing a 200+ lines transpilers and, trust me, based on my experience comments are needed if you want to PR it, because it's not the common event patch. But again, this is not the case.
Last thing, I don't want to know the reason, but you've made an event to compute the Scp330 pickup or something, I've seen it somewhere, do I really have to say anything about that? You're literally copy-pasting the base-game code, and it doesn't make sense. Helper methods are a thing, true, but let's not use helper method with IL just by calling them once and make them doing all the work like if it was a prefix, because it kills the meaning of writing transpilers.

**Comment by Undid-Iridium** [comment]
I don't feel like reviewing all the patches, because there're the same issues as the reviewed ones. Also, I don't really find necessary describing and/or commenting every IL step as it's probably made from and for who's writing them. Counting the stack is useless, because you'd have to maintain all the comments, this means that you have to update transpilers when needed, since SL-12.0 is not so far away, and you also have to update comments in order to make them coherent with the written code.
Since I'll be the one who's gonna update them, I don't want to lose my time at updating comments made by someone which is clearly over-commenting everything.
That said, it's not useless commenting code, but that's an over-commenting case, which is bad.
You're also breaking style patterns by adding curly brackets in the for loops where you're yield returning the new instructions. Let's just keep everything as more uniform as possible, right?
Commenting IL code is a double-edged sword, because you should only comment logical steps rather than talking about "original {INSERT_SOMETHING_HERE} logic" or something like that, because as I said before, we're not here to maintain a book written by Shakespeare; maintaining the code is already enough in my opinion.
I appreciate the fact that you're documenting IL, which is needed sometimes, but it's not the right way to document IL; you're not writing a 200+ lines transpilers and, trust me, based on my experience comments are needed if you want to PR it, because it's not the common event patch. But again, this is not the case.
Last thing, I don't want to know the reason, but you've made an event to compute the Scp330 pickup or something, I've seen it somewhere, do I really have to say anything about that? You're literally copy-pasting the base-game code, and it doesn't make sense. Helper methods are a thing, true, but let's not use helper method with IL just by calling them once and make them doing all the work like if it was a prefix, because it kills the meaning of writing transpilers.

1. I don't feel like reviewing all the patches, because there're the same issues as the reviewed ones. 
1A. I do not know what you mean by this, I was under the impression the reviewed prefixs were fine, except they needed to be translated. 

2. Also, I don't really find necessary describing and/or commenting every IL step as it's probably made from and for who's writing them. Counting the stack is useless, because you'd have to maintain all the comments, this means that you have to update transpilers when needed, since SL-12.0 is not so far away, and you also have to update comments in order to make them coherent with the written code.
2A. Ok, I was trying to make it where people can learn the logic behind it, but sure.

3.Since I'll be the one who's gonna update them, I don't want to lose my time at updating comments made by someone which is clearly over-commenting everything.
3A. Ok

4. That said, it's not useless commenting code, but that's an over-commenting case, which is bad.
4A. Ok

5. You're also breaking style patterns by adding curly brackets in the for loops where you're yield returning the new instructions. Let's just keep everything as more uniform as possible, right?
5A. I copied pasted from somewhere, this is a format change. I would suggest then adding rules to stylecop to prevent this kind of confliction. 

6. Commenting IL code is a double-edged sword, because you should only comment logical steps rather than talking about "original {INSERT_SOMETHING_HERE} logic" or something like that, because as I said before, we're not here to maintain a book written by Shakespeare; maintaining the code is already enough in my opinion.
I appreciate the fact that you're documenting IL, which is needed sometimes, but it's not the right way to document IL; you're not writing a 200+ lines transpilers and, trust me, based on my experience comments are needed if you want to PR it, because it's not the common event patch. But again, this is not the case.

6A. Ok, I've seen patches without any comments, and those with some comments but C# based. I would presume you would prefer: 
If you are going to comment, comment in C# generalized idea of what is going on. Otherwise, do not comment. 

7. Last thing, I don't want to know the reason, but you've made an event to compute the Scp330 pickup or something, I've seen it somewhere, do I really have to say anything about that? You're literally copy-pasting the base-game code, and it doesn't make sense. Helper methods are a thing, true, but let's not use helper method with IL just by calling them once and make them doing all the work like if it was a prefix, because it kills the meaning of writing transpilers.

7A. Yamato wrote it to solve a problem where candies were generated randomly rather than potentially set by event. If this is incorrect then it is a matter of redirecting the call to the original function, and doing a transpiler there. If that is still incorrect and the original logic should be left alone then let me know so I can just rewrite the transpiler to call the original way/allow the original code to do the call.

File: Exiled.Events/Patches/Events/Scp244/PickingUpScp244.cs
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp244/PickingUpScp244.cs:74]: I need to be able to go over the return. 

```
// If isAllowed = 1, jump to continue route, otherwise, return occurs below
                new(OpCodes.Brtrue, continueProcessing),
 // False Route
                new CodeInstruction(OpCodes.Ret),
 // Good route of is allowed being true
                new CodeInstruction(OpCodes.Nop).WithLabels(continueProcessing),
```

How else would you prefer I go over it? I mean, the only other way is to calculate the size of the instructions I am adding, then calculate the position I would normally be at ( The next IL code) and then add a label to it. Is that the preferred way of doing it?

File: Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs:47]: This was for clarification beyond just git blame, to know who tested it, and who to yell at - I can remove it since it is exasperating to you.

File: Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs:86]: Is this not short form?

File: Exiled.Events/Patches/Events/Scp244/UsingScp244.cs
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp244/UsingScp244.cs:42]: Will update, https://github.com/Exiled-Team/EXILED/pull/1192#discussion_r862506285

File: Exiled.Events/Patches/Events/Scp244/UsingScp244.cs
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp244/UsingScp244.cs:77]: https://github.com/Exiled-Team/EXILED/pull/1192#discussion_r862506348

File: Exiled.Events/Patches/Events/Scp244/UsingScp244.cs
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp244/UsingScp244.cs:80]: https://github.com/Exiled-Team/EXILED/pull/1192#discussion_r862506184

File: Exiled.Events/Patches/Events/Scp330/DroppingCandy.cs
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp330/DroppingCandy.cs:53]: https://github.com/Exiled-Team/EXILED/pull/1192#discussion_r862506285

File: Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs:86]: Oh, did you mean the CLT? If so, yes, I need to short form that

**Comment by NaoUnderscore** [comment]
> 1. I don't feel like reviewing all the patches, because there're the same issues as the reviewed ones.
>        1A. I do not know what you mean by this, I was under the impression the reviewed prefixs were fine, except they needed to be translated.
> 
>     2. Also, I don't really find necessary describing and/or commenting every IL step as it's probably made from and for who's writing them. Counting the stack is useless, because you'd have to maintain all the comments, this means that you have to update transpilers when needed, since SL-12.0 is not so far away, and you also have to update comments in order to make them coherent with the written code.
>        2A. Ok, I was trying to make it where people can learn the logic behind it, but sure.
> 
>     3. Since I'll be the one who's gonna update them, I don't want to lose my time at updating comments made by someone which is clearly over-commenting everything.
>        3A. Ok
> 
>     4. That said, it's not useless commenting code, but that's an over-commenting case, which is bad.
>        4A. Ok
> 
>     5. You're also breaking style patterns by adding curly brackets in the for loops where you're yield returning the new instructions. Let's just keep everything as more uniform as possible, right?
>        5A. I copied pasted from somewhere, this is a format change. I would suggest then adding rules to stylecop to prevent this kind of confliction.
> 
>     6. Commenting IL code is a double-edged sword, because you should only comment logical steps rather than talking about "original {INSERT_SOMETHING_HERE} logic" or something like that, because as I said before, we're not here to maintain a book written by Shakespeare; maintaining the code is already enough in my opinion.
>        I appreciate the fact that you're documenting IL, which is needed sometimes, but it's not the right way to document IL; you're not writing a 200+ lines transpilers and, trust me, based on my experience comments are needed if you want to PR it, because it's not the common event patch. But again, this is not the case.
>        6A. Ok, I've seen patches without any comments, and those with some comments but C# based. I would presume you would prefer:
>        If you are going to comment, comment in C# generalized idea of what is going on. Otherwise, do not comment.
> 
>     7. Last thing, I don't want to know the reason, but you've made an event to compute the Scp330 pickup or something, I've seen it somewhere, do I really have to say anything about that? You're literally copy-pasting the base-game code, and it doesn't make sense. Helper methods are a thing, true, but let's not use helper method with IL just by calling them once and make them doing all the work like if it was a prefix, because it kills the meaning of writing transpilers.
>        7A. Yamato wrote it to solve a problem where candies were generated randomly rather than potentially set by event. If this is incorrect then it is a matter of redirecting the call to the original function, and doing a transpiler there. If that is still incorrect and the original logic should be left alone then let me know so I can just rewrite the transpiler to call the original way/allow the original code to do the call.
> 
> 
> Github formatting is being annoying.

1. Conventions and optimization, that's what I'm talking about. Short form is enough since you're not jumping tons of bytes.

2. As I said before, we're not here to teach people IL nor transpilers/Harmony; I appreciate you got it.

6. Patches without any comments have been probably made by me, but not only. Since everyone is now enough comfortable at writing IL, regarding the dev team (Me, Joker and some contributors between who actively contributes), we don't usually comment IL anymore (I started at writing documented IL but then I dropped that habit because of the reason I explained before). Unless it's really necessary, like a 200+ lines IL transpiler, or some weird transpiler which does some weird jumps whereas other transpilers are much easier to read and/or write, we don't really feel it's necessary commenting everything then.

7. Pretty sure there's another route, hence I kindly ask y'all to find and follow it.

**Comment by Undid-Iridium** [comment]
> > 1. I don't feel like reviewing all the patches, because there're the same issues as the reviewed ones.
> >    1A. I do not know what you mean by this, I was under the impression the reviewed prefixs were fine, except they needed to be translated.
> >    
> >    2. Also, I don't really find necessary describing and/or commenting every IL step as it's probably made from and for who's writing them. Counting the stack is useless, because you'd have to maintain all the comments, this means that you have to update transpilers when needed, since SL-12.0 is not so far away, and you also have to update comments in order to make them coherent with the written code.
> >       2A. Ok, I was trying to make it where people can learn the logic behind it, but sure.
> >    3. Since I'll be the one who's gonna update them, I don't want to lose my time at updating comments made by someone which is clearly over-commenting everything.
> >       3A. Ok
> >    4. That said, it's not useless commenting code, but that's an over-commenting case, which is bad.
> >       4A. Ok
> >    5. You're also breaking style patterns by adding curly brackets in the for loops where you're yield returning the new instructions. Let's just keep everything as more uniform as possible, right?
> >       5A. I copied pasted from somewhere, this is a format change. I would suggest then adding rules to stylecop to prevent this kind of confliction.
> >    6. Commenting IL code is a double-edged sword, because you should only comment logical steps rather than talking about "original {INSERT_SOMETHING_HERE} logic" or something like that, because as I said before, we're not here to maintain a book written by Shakespeare; maintaining the code is already enough in my opinion.
> >       I appreciate the fact that you're documenting IL, which is needed sometimes, but it's not the right way to document IL; you're not writing a 200+ lines transpilers and, trust me, based on my experience comments are needed if you want to PR it, because it's not the common event patch. But again, this is not the case.
> >       6A. Ok, I've seen patches without any comments, and those with some comments but C# based. I would presume you would prefer:
> >       If you are going to comment, comment in C# generalized idea of what is going on. Otherwise, do not comment.
> >    7. Last thing, I don't want to know the reason, but you've made an event to compute the Scp330 pickup or something, I've seen it somewhere, do I really have to say anything about that? You're literally copy-pasting the base-game code, and it doesn't make sense. Helper methods are a thing, true, but let's not use helper method with IL just by calling them once and make them doing all the work like if it was a prefix, because it kills the meaning of writing transpilers.
> >       7A. Yamato wrote it to solve a problem where candies were generated randomly rather than potentially set by event. If this is incorrect then it is a matter of redirecting the call to the original function, and doing a transpiler there. If that is still incorrect and the original logic should be left alone then let me know so I can just rewrite the transpiler to call the original way/allow the original code to do the call.
> > 
> > Github formatting is being annoying.
> 
> `1.` Conventions and optimization, that's what I'm talking about. Short form is enough since you're not jumping tons of bytes.
> 
> `2.` As I said before, we're not here to teach people IL nor transpilers/Harmony; I appreciate you got it.
> 
> `6.` Patches without any comments have been probably made by me, but not only. Since everyone is now enough comfortable at writing IL, regarding the dev team (Me, Joker and some contributors between who actively contributes), we don't usually comment IL anymore (I started at writing documented IL but then I dropped that habit because of the reason I explained before). Unless it's really necessary, like a 200+ lines IL transpiler, or some weird transpiler which does some weird jumps whereas other transpilers are much easier to read and/or write, we don't really feel it's necessary commenting everything then.
> 
> `7.` Pretty sure there's another route, hence I kindly ask y'all to find and follow it.

1. Understood, thanks
2. Sure - I understand and will follow suite. 
6. Understood, thanks
7. I will work with Yamato (He's out for a week) on this. I may just think of a solution myself (as I first need to figure out exactly what he was trying to do). 

Appreciate the feedback, thanks!

File: Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs:86]: Nevermind, I understand what you mean now. You meant a short form jump, not reduction of Code Instruction to short form C#9. Tiny brain moment

**Comment by louis1706** [comment]
I just want to say you can PR a change on this branch and i will merge it instantly if it's one of you who send it :3

**Comment by Undid-Iridium** [comment]
@NaoUnderscore Ok, I have updated the code to be more simplistic. However, I understand now what @louis1706 was trying to do for ServerProcessPickup. To simplify, Event handler has extra information that the original function does not use. Therefore, he adds the ability to specify the candy (What if developer wanted everyone to only get pink candies, or blue, etc.). Furthermore, the original code had an inefficient loop

```
else
			{
				while (pickup.StoredCandies.Count > 0 && bag.TryAddSpecific(pickup.StoredCandies[0]))
				{
					result = true;
					pickup.StoredCandies.RemoveAt(0);
				}
			}
```

Which in itself is not an issue but is rather just bad. If you prefer to not use a custom function that avoids some of the original codes pitfalls then I can easily just make a new pickup object, add the specific candy the events wants, and call it a day. I just want a confirmation of what is preferable before I go down a route. Thanks

**Comment by Undid-Iridium** [comment]
Also @louis1706 I already have a branch, and pr with the minimal code changes required to reduce nop, and ret calls. I will update that with a "merge me" when I am done, and when I can confirm whether to rewrite InteractingScp330 code based on my previous comment. Thanks

**Comment by Undid-Iridium** [comment]
FYI, the branch @louis1706 merged is not done. It was merged before any change to the code. For simplicity sake, I am just going to make the modification tomorrow to create a new Scp330Pickup with the specific candy(s) to be added.

File: Exiled.Events/Patches/Events/Scp244/PickingUpScp244.cs
Comment (NaoUnderscore) [reply Exiled.Events/Patches/Events/Scp244/PickingUpScp244.cs]: `Brfalse.s, ret` where `ret` is a label added to the last instruction.

File: Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs
Comment (NaoUnderscore) [reply Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs]: I mean, we can't really add comments for everything that has been tested and/or who tested it. You feel me?

File: Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs
Comment (NaoUnderscore) [reply Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs:74]: 👍🏼 The short for is the normal form but it can jump less

**Comment by NaoUnderscore** [comment]
> 

I'd just rewrite it in IL, so it's up to you kekw.

**Comment by louis1706** [comment]
Nao can you also assign irriduim for this PR with me

File: Exiled.Events/Patches/Events/Scp244/PickingUpScp244.cs
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp244/PickingUpScp244.cs]: Yep, did that. Brain kicked in in terms of reversing the conditions and calling the last return.

File: Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp244/UpdateScp244.cs]: Sure, I understand - no worries.

**Comment by Undid-Iridium** [comment]
I'll do the simplistic route first, make sure it works, then try to transpiler ServerProcessPickup (There are several spots I would have to update/remove so that won't be a quick effort as far as I can tell). Also, I do not need to be assigned, Louis is sufficient as this is their PR.

**Comment by louis1706** [comment]
> I'll do the simplistic route first, make sure it works, then try to transpiler ServerProcessPickup (There are several spots I would have to update/remove so that won't be a quick effort as far as I can tell). Also, I do not need to be assigned, Louis is sufficient as this is their PR. 

You have made all the IL Code it's why i have requests it ^^

**Comment by Undid-Iridium** [comment]
So.. my attempt to go down the "simplistic" route was a mistake. It seems that Scp330Pickup is something that I cannot just instantiate and pass (I can create the object, verify it is not null, but upon calling a function with it, it turns null. I would presume this is because there is a lot more steps than just creating a new object, such as ServerAddItem). Therefore, it becomes more than just a overwrite the original function issue, as I cannot pass a non-null Scp330Pickup object. Consequently, I then cannot use the original function for the needed functionality. Thus, calling a custom function seems to be the correct choice. I would have to say the current way of handling adding specific candies seems correct with a custom function where the candy is provided and a Scp330Pickup is not created. While we are using a transpiler to call a custom function, in this case I believe it is necessary as patching the original function is of no benefit since we cannot pass the parameter with the changes needed anyway. In conclusion, either I have a skill issue or the NW approach is very throttling and to avoid it is to create a new function, @NaoUnderscore since I defaulting back to the original approach from @louis1706 - let me know what you think/want instead. Thanks

File: Exiled.Events/Patches/Events/Scp244/DamagingScp244.cs
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp244/DamagingScp244.cs:50]: No, do not disable it. Just stop leaving a blank line after every instruction.

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Events/Scp244/DamagingScp244.cs
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp244/DamagingScp244.cs:50]: image

Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp244/DamagingScp244.cs:50]: image

Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp244/DamagingScp244.cs:50]: I mean.. the other Transpilers themselves have the same disabling, spacing does not resolve the type of issue which is multi-line parameter.

Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp244/DamagingScp244.cs:50]: I can move it up, but going by precedent, it should exist in the file. Alternatively, make it a global configured ignored as multi-line parameter is used for most Transpilers. Up to you

Comment (NaoUnderscore) [reply Exiled.Events/Patches/Events/Scp244/DamagingScp244.cs:50]: Oh lol, move it up then. And yeah, stop leaving blank lines. That's all 👍🏼

Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Scp244/DamagingScp244.cs:50]: Done, on @louis1706 to merge.

Comment (louis1706) [reply Exiled.Events/Patches/Events/Scp244/DamagingScp244.cs]: Done ping me discord :) when you want to merge i will see it faster than GitHub

**Review by joker-119 — APPROVED** [review]

**Pull request merged**
